receiver/k8sobjectsreceiver/config.go

```diff
@@ -50,26 +59,61 @@ type K8sObjectsConfig struct {
 type Config struct {
 	k8sconfig.APIConfig `mapstructure:",squash"`
 
-	Objects []*K8sObjectsConfig `mapstructure:"objects"`
+	Objects   []*K8sObjectsConfig `mapstructure:"objects"`
+	ErrorMode ErrorMode           `mapstructure:"error_mode"`
+	logger    *zap.Logger
 
 	// For mocking purposes only.
 	makeDiscoveryClient func() (discovery.ServerResourcesInterface, error)
 	makeDynamicClient   func() (dynamic.Interface, error)
 }
 
 func (c *Config) Validate() error {
+	zapCfg := zap.NewProductionConfig()
+	logger, err := zapCfg.Build()
+	if err != nil {
+		return fmt.Errorf("failed to create logger: %w", err)
+	}
+	c.logger = logger
```

Comment:
please do not create a logger or change the state of the Config struct in Validate.